FinSearchUnified/Tests/Helper/UrlBuilderTest.php

```diff
@@ -150,4 +177,183 @@ private function setIpHeader($field, $ipAddress)
     {
         $_SERVER[$field] = $ipAddress;
     }
+
+    public function successfulRequestProvider()
+    {
+        return [
+            'Request was successful with DI enabled' => ['{"directIntegration":{"enabled":1}}', true],
+            'Request was successful with DI disabled' => ['{"directIntegration":{"enabled":0}}', false]
+        ];
+    }
+
+    public function unsuccessfulRequestProvider()
+    {
+        return [
+            'Request was not successful with DI disabled' => ['API', false],
+            'Request was not successful with DI enabled' => ['Direct Integration', true]
+        ];
+    }
+
+    public function exceptionOnRequestProvider()
+    {
+        return [
+            'Request causes "Zend_Http_Client_Exception" with DI enabled' => ['Direct Integration', true],
+            'Request causes "Zend_Http_Client_Exception" with DI disabled' => ['API', false]
+        ];
+    }
+
+    /**
+     * @dataProvider successfulRequestProvider
+     *
+     * @param string $responseBody
+     * @param bool $expectedStatus
+     *
+     * @throws \Zend_Http_Exception
+     * @throws \Exception
+     */
+    public function testConfigStatusOnSuccessfulRequest($responseBody, $expectedStatus)
+    {
+        $configArray = [
+            ['IntegrationType', Constants::INTEGRATION_TYPE_DI]
+        ];
+
+        // Create Mock object for Shopware Config
+        $config = $this->getMockBuilder(Shopware_Components_Config::class)
+            ->setMethods(['offsetGet'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $config->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($configArray);
+
+        // Assign mocked config variable to application container
+        Shopware()->Container()->set('config', $config);
+
+        $this->httpResponse = new Zend_Http_Response(200, [], $responseBody);
+
+        $httpClientMock = $this->getMockBuilder(Zend_Http_Client::class)
+            ->setMethods(['request'])
+            ->getMock();
+        $httpClientMock->expects($this->once())
+            ->method('request')
+            ->willReturn($this->httpResponse);
+
+        $this->httpClient = $httpClientMock;
+
+        $urlBuilder = new UrlBuilder($this->httpClient);
+
+        $status = $urlBuilder->getConfigStatus();
+
+        $this->assertEquals(
+            $expectedStatus,
+            $status,
+            sprintf(
+                'Expected config status to return %s ',
```

Comment:
* Try to find a more specific assertion message e.g. `Expected integration type to be DI...` or similar.
* Remove the trailing whitespace.